src/lib/operators/table_scan.hpp

```diff
@@ -28,6 +40,127 @@ class TableScan : public AbstractOperator {
 
  protected:
   std::shared_ptr<const Table> _on_execute() override;
+
+  ColumnID _column_id;
+  ScanType _scan_type;
+  AllTypeVariant _search_value;
+
+  std::shared_ptr<const BaseTableScanImpl> _table_scan_impl;
+
+  template<typename T>
+  class TableScanImpl : public BaseTableScanImpl {
+   public:
+    TableScanImpl(ColumnID column_id, const ScanType scan_type,
+            const AllTypeVariant search_value, std::shared_ptr<const Table> input_table ) : _column_id(column_id), _scan_type(scan_type), _search_value(type_cast<T>(search_value)), _input_table(input_table) {}
+    
+    const std::function<bool(const T&)> compare_lambda(ScanType comp) const {
```

Comment:
Maybe call this something like `create_compare_lambda` ? And as funnction outside of class perhaps as it doesn't depend on the state of the class but is more like a helper function.